micropython/drivers/imu/lsm9ds1/lsm9ds1.py

```diff
@@ -58,129 +59,149 @@
-        self.scale_accel = 32768 / _SCALE_ACCEL[scale_accel][0]
-
-    def init_magnetometer(self, sample_rate=7, scale_magnet=0):
-        """
-        sample rates = 0-7 (0.625, 1.25, 2.5, 5, 10, 20, 40, 80Hz)
-        scaling = 0-3 (+/-4, +/-8, +/-12, +/-16 Gauss)
-        """
-        assert sample_rate < 8, "invalid sample rate: %d (0-7)" % sample_rate
-        assert scale_magnet < 4, "invalid scaling: %d (0-3)" % scale_magnet
-        i2c = self.i2c
-        addr = self.address_magnet
-        mv = memoryview(self.scratch)
-        mv[0] = 0x40 | (sample_rate << 2)  # ctrl1: high performance mode
-        mv[1] = scale_magnet << 5  # ctrl2: scale, normal mode, no reset
+        # Configure Magnetometer
+        mv[0] = 0x40 | (magnet_odr << 2)  # ctrl1: high performance mode
+        mv[1] = MAGNET_SCALE.index(magnet_scale) << 5  # ctrl2: scale, normal mode, no reset
         mv[2] = 0x00  # ctrl3: continous conversion, no low power, I2C
         mv[3] = 0x08  # ctrl4: high performance z-axis
         mv[4] = 0x00  # ctr5: no fast read, no block update
-        i2c.writeto_mem(addr, _CTRL_REG1_M, mv[:5])
-        self.scale_factor_magnet = 32768 / ((scale_magnet + 1) * 4)
+        self.bus.writeto_mem(self.address_magnet, _CTRL_REG1_M, mv[:5])
+
+        self.gyro_scale = 32768 / gyro_scale
+        self.accel_scale = 32768 / accel_scale
+        self.scale_factor_magnet = 32768 / ((MAGNET_SCALE.index(magnet_scale) + 1) * 4)
+
+        # Allocate scratch buffer for efficient conversions and memread op's
+        self.scratch_int = array.array("h", [0, 0, 0])
 
     def calibrate_magnet(self, offset):
         """
-        offset is a magnet vecor that will be substracted by the magnetometer
+        offset is a magnet vector that will be subtracted by the magnetometer
         for each measurement. It is written to the magnetometer's offset register
         """
         offset = [int(i * self.scale_factor_magnet) for i in offset]
-        mv = memoryview(self.scratch)
-        mv[0] = offset[0] & 0xFF
-        mv[1] = offset[0] >> 8
-        mv[2] = offset[1] & 0xFF
-        mv[3] = offset[1] >> 8
-        mv[4] = offset[2] & 0xFF
-        mv[5] = offset[2] >> 8
-        self.i2c.writeto_mem(self.address_magnet, _OFFSET_REG_X_M, mv[:6])
+        cal = bytearray(6)
+        cal[0] = offset[0] & 0xFF
+        cal[1] = offset[0] >> 8
+        cal[2] = offset[1] & 0xFF
+        cal[3] = offset[1] >> 8
+        cal[4] = offset[2] & 0xFF
+        cal[5] = offset[2] >> 8
+        self.bus.writeto_mem(self.address_magnet, _OFFSET_REG_X_M, cal)
 
     def gyro_id(self):
-        return self.i2c.readfrom_mem(self.address_gyro, _WHO_AM_I, 1)
+        return self.bus.readfrom_mem(self.address_imu, _WHO_AM_I, 1)
 
     def magent_id(self):
-        return self.i2c.readfrom_mem(self.address_magnet, _WHO_AM_I, 1)
+        return self.bus.readfrom_mem(self.address_magnet, _WHO_AM_I, 1)
 
-    def magnet(self):
+    def read_magnet(self):
```

Comment:
You've renamed these methods to add `read_` at the start.  But there are existing methods in other drivers like `altitude()` and `pressure()` (also see below in the hs3003 driver).

For consistency they should either all start with `read_`, or none should start with `read_`.

Since we don't want to make breaking changes to existing method names, please keep these names as-is, and make sure the newly added drivers also don't start with `read_` for their methods.

Reply:
> Since we don't want to make breaking changes to existing method names

The lsm6dsox driver already used `read_` so I followed that one because it's newer,  the lsm9 driver doesn't, so I changed it to match, either way there will be breaking changes.